nossas/bonde
# Feature user retrieve password

# Related issues
- Add page to retrieve password #659

# Related pull-requests
- Issue 234 - Create new password for user nossas/bonde-server#203

# How to test
- Access the [login page](http://app.staging.bonde.org/login)
- Entry an invalid password
- It should display the unauthorized error like screenshot below:
<img width="528" alt="screen shot 2017-06-13 at 04 35 06" src="https://user-images.githubusercontent.com/5435389/27071026-b5088b00-4ff1-11e7-9ef3-c449d0793f02.png">

- Click on **ESQUECEU SUA SENHA?** link on the password input label
- It should redirect to `/account/retrieve` route
- The account password retrieve page should looks like screenshot below:
<img width="1055" alt="screen shot 2017-06-13 at 04 36 46" src="https://user-images.githubusercontent.com/5435389/27071105-eec34718-4ff1-11e7-8d5a-cff738a710fc.png">

- Check the input validations:
  - Try to submit with input as empty
    - It should show the error below:

<img width="459" alt="screen shot 2017-06-13 at 04 37 46" src="https://user-images.githubusercontent.com/5435389/27071148-1acfc944-4ff2-11e7-8c26-c183eddc6da4.png">

  - Try to entry an email that have an invalid format
    - It should show the error below:
<img width="468" alt="screen shot 2017-06-13 at 04 38 37" src="https://user-images.githubusercontent.com/5435389/27071170-358b1e96-4ff2-11e7-8ca3-b79544dd0084.png">

- Entry a valid email address and click on **ENVIAR** button or, hit enter
- It should redirect to login page
- It should show a notification that an email has been sent with new password
<img width="1334" alt="screen shot 2017-06-13 at 04 42 20" src="https://user-images.githubusercontent.com/5435389/27071304-baf57c52-4ff2-11e7-9976-7fe20c9ee7d8.png">
